### PR TITLE
feat(repository): remove multiple db calls for find by id

### LINKED ISSUE #87: multiple DB calls while using findById
FindById has the below statement which can be reduced by checking the deleted flag and in turn improving performance 
 const entityToRemove = await super.findOne(filter, options);
        if (entityToRemove) {
            // Now call super
            return super.findById(id, filter, options);
        }
        else {
            throw new rest_1.HttpErrors.NotFound("EntityNotFound" /* ErrorKeys.EntityNotFound */);
        }

### PR DESCRIPTION
**Pull request by yeshamavani**
#87

## Description

While using findById there are two database calls causing latency.
combined those into one.

Fixes #87

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## Checklist:

- [ ] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated

### CODE REVIEW

**Review by akshatdubeysf — APPROVED** [review]

File: src/repositories/soft-crud.repository.base.ts [hunk 1]
```diff
@@ -194,13 +189,9 @@ export abstract class SoftCrudRepository<
...
   ): Promise<T & Relations> {
-    //As parent method findById have filter: FilterExcludingWhere<T>
-    //so we need add check here.
-    const entityToRemove = await super.findOne(filter, options);
-
-    if (entityToRemove) {
-      // Now call super
-      return super.findById(id, filter, options);
+    const entity = await super.findById(id, filter, options);
+    if (entity) {
```
Comment (mlakdawala-sf) [inline src/repositories/soft-crud.repository.base.ts:193]:
```suggestion
 if (entity && !entity.deleted) {
```

Comment (yeshamavani) [reply src/repositories/soft-crud.repository.base.ts:193]: here we are returning even the soft deleted entity hence not added.
is that fine?

Comment (mlakdawala-sf) [reply src/repositories/soft-crud.repository.base.ts:193]: got it

**Review by mlakdawala-sf — APPROVED** [review]

**Pull request merged**